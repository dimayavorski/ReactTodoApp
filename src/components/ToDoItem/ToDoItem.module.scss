.toDoItem {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    background-color: white;
    padding: 5px 15px 5px 15px;
    gap: 15px;
    border-bottom: 1px solid gray;

    .toDoItemInfo {
        display: flex;

    }

    .removeBtn {
        border: none;
        background: none;
        cursor:pointer;
    }
}
    