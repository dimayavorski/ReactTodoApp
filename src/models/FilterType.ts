export enum FilterType {
	ALL = 0,
	COMPLETED = 1,
	INCOMPLETE = 2,
}
