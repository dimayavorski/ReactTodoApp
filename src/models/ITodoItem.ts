export interface ITodoItem {
	id: string;
	text: string;
	checked: boolean;
}
