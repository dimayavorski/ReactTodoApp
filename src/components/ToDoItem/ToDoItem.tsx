import { ITodoItem } from '../../models/ITodoItem';
import styles from './ToDoItem.module.scss';
import { FC } from 'react';
import { AiOutlineClose } from 'react-icons/ai';
import { Checkbox } from '../Checkbox/Checkbox';
import { useTodos } from '../../hooks/useTodos';

interface TodoItemProps {
	todoItem: ITodoItem;
}

export const ToDoItem: FC<TodoItemProps> = ({ todoItem }) => {
	const { remove, toggleComplete } = useTodos();

	return (
		<div className={styles.toDoItem}>
			<div
				className={
					todoItem.checked ? styles.toDoItemInfoChecked : styles.toDoItemInfo
				}
			>
				<Checkbox
					isActive={todoItem.checked}
					toggleCheckboxHandler={() => toggleComplete(todoItem)}
				/>
				<p>{todoItem.text}</p>
			</div>
			<button className={styles.removeBtn} onClick={() => remove(todoItem)}>
				<AiOutlineClose size={20} />
			</button>
		</div>
	);
};
