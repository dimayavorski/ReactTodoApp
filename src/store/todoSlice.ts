import { PayloadAction, createSlice } from '@reduxjs/toolkit';
import { ITodoItem } from '../models/ITodoItem';
import { FilterType } from '../models/FilterType';

const todos: ITodoItem[] = [];
const filter: number = FilterType.ALL;

const todoSlice = createSlice({
	name: 'todos',
	initialState: {
		todos,
		filter,
	},
	reducers: {
		addToDo(state, action: PayloadAction<ITodoItem>) {
			state.todos.push(action.payload);
		},
		removeToDo(state, action: PayloadAction<ITodoItem>) {
			state.todos = state.todos.filter((todo) => todo.id !== action.payload.id);
		},
		toggleToDoComplete(state, action: PayloadAction<ITodoItem>) {
			const toggleTodo = state.todos.find(
				(todo) => todo.id === action.payload.id
			);
			if (toggleTodo) toggleTodo.checked = !toggleTodo.checked;
		},
		updateToDos(state, action: PayloadAction<ITodoItem[]>) {
			state.todos = action.payload;
		},
		updateFilter(state, action: PayloadAction<FilterType>) {
			state.filter = action.payload;
		},
	},
});

export const {
	updateToDos,
	addToDo,
	removeToDo,
	toggleToDoComplete,
	updateFilter,
} = todoSlice.actions;

export default todoSlice.reducer;
