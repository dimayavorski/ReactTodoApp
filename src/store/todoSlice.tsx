import { PayloadAction, createSlice } from '@reduxjs/toolkit';
import { ITodoItem } from '../models/ITodoItem';

const todos: ITodoItem[] = [];

const todoSlice = createSlice({
	name: 'todos',
	initialState: {
		todos,
	},
	reducers: {
		addToDo(state, action: PayloadAction<ITodoItem>) {
			state.todos.push(action.payload);
		},
		removeToDo(state, action: PayloadAction<ITodoItem>) {
			state.todos = state.todos.filter((todo) => todo.id !== action.payload.id);
		},
		toggleToDoComplete(state, action: PayloadAction<ITodoItem>) {
			const toggleTodo = state.todos.find(
				(todo) => todo.id === action.payload.id
			);
			if (toggleTodo) toggleTodo.checked = !toggleTodo.checked;
		},
		updateToDos(state, action: PayloadAction<ITodoItem[]>) {
			state.todos = action.payload;
		},
	},
});

export const { updateToDos, addToDo, removeToDo, toggleToDoComplete } =
	todoSlice.actions;
//export const selectCount = (state: RootState) => state.counter.value
export default todoSlice.reducer;
