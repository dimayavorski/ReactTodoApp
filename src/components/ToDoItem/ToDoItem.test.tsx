import { ITodoItem } from '../../models/ITodoItem';
import {
	RenderResult,
	fireEvent,
	render,
	screen,
} from '@testing-library/react';
import { ToDoItem } from './ToDoItem';
import { Provider } from 'react-redux';
import { store } from '../../store/store';

const mockItem: ITodoItem = {
	id: 'asdasd',
	text: 'someName',
	checked: true,
};

let component: RenderResult;

describe('ToDoItem tests', () => {
	beforeEach(() => {
		component = render(
			<Provider store={store}>
				<ToDoItem todoItem={mockItem} />
			</Provider>
		);
	});

	it('should match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should render component', () => {
		expect(screen.getByText(mockItem.text)).toBeInTheDocument();
	});

	it('should change checkbox value', () => {
		const checkbox = screen.getByRole('checkbox');

		fireEvent.change(checkbox, { target: { checked: true } });

		expect(checkbox).toBeChecked();
	});
});
