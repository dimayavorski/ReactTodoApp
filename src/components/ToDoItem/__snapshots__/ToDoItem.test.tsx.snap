// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoItem tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul>
        <div
          class="toDoItem"
        >
          <div
            class="toDoItemInfoChecked"
          >
            <div
              class="checkBoxContainer"
              data-testid="checkBoxContainer"
            >
              <label>
                <input
                  checked=""
                  type="checkbox"
                />
                <span />
              </label>
            </div>
            <p>
              someName
            </p>
          </div>
          <button
            class="removeBtn"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </button>
        </div>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul>
      <div
        class="toDoItem"
      >
        <div
          class="toDoItemInfoChecked"
        >
          <div
            class="checkBoxContainer"
            data-testid="checkBoxContainer"
          >
            <label>
              <input
                checked=""
                type="checkbox"
              />
              <span />
            </label>
          </div>
          <p>
            someName
          </p>
        </div>
        <button
          class="removeBtn"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </button>
      </div>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
